chef/automate
# [auth-2624] user modal update

### :nut_and_bolt: Description: What code changed, and why?

When did this...
![image](https://user-images.githubusercontent.com/21015366/72644673-6e486480-3926-11ea-97d5-eeea646d43c0.png)

Become hotter than this???
<img width="1081" alt="Screen Shot 2020-01-17 at 12 39 59 PM" src="https://user-images.githubusercontent.com/21015366/72644699-861fe880-3926-11ea-8f9e-95a29a21eb3b.png">


This PR updates the user creation modal to be consistent with the token/team/project creation modals.

Instead of making the `create-object-modal` fit with the special case of users, we decided to keep the user modal a part of the `user-management` component. We decided it would over-complicate the `create-object-modal` to add special casing for the user modal (extra password fields, rename Name and ID fields to Display Name and Username). This decision resulted in some duplicated code which we should probably extract into a common helper library once we add another creation modal.

There is some potential to refactor the `create-object-modal` so that we can slot any form we need into it. We might consider this when we get to adding a role creation modal (which will need a dropdown for actions). Then we could revisit using the `create-object-modal` on the user page.

### :+1: Definition of Done
- [x] modal styles match [zeplin mock](https://zpl.io/25JAppn)
- [x] conflict and bad password errors are handled inside the modal (fixes https://github.com/chef/automate/issues/2401)
- [x] the Display Name is used to auto-generate the Username

### :athletic_shoe: How to Build and Test the Change
- start up the UI however you like to
- navigate to https://a2-dev.test/settings/users
- select `Create User` and try creating some users with the snazzy new modal
- create a user with the same username as another user (like `editor` or `admin`) and see the error in the modal
- create a user with the password `aaaaaaaa` and see the error in the modal
- create a user with an auto-generated username
- create a user with a custom username

### :white_check_mark: Checklist
